packages/cli/src/processed-data-managers/processed-data-helper.ts

```diff
@@ -0,0 +1,322 @@
+import { createHash } from 'crypto';
+import { ApplicationError } from 'n8n-workflow';
+import type {
+	ICheckProcessedContextData,
+	IProcessedDataManager,
+	ICheckProcessedOptions,
+	ICheckProcessedOutput,
+	ProcessedDataContext,
+	ProcessedDataItemTypes,
+	ProcessedDataMode,
+} from 'n8n-workflow';
+import { Container } from 'typedi';
+
+import { ProcessedDataRepository } from '@/databases/repositories/processed-data.repository';
+import type { IProcessedDataEntries, IProcessedDataLatest } from '@/interfaces';
+
+export class ProcessedDataHelper implements IProcessedDataManager {
+	private static sortEntries(
+		items: ProcessedDataItemTypes[],
+		mode: ProcessedDataMode,
+	): ProcessedDataItemTypes[] {
+		return [...items].sort((a, b) => (ProcessedDataHelper.compareValues(mode, a, b) ? 1 : -1));
+	}
+
+	private static compareValues(
+		mode: ProcessedDataMode,
+		value1: ProcessedDataItemTypes,
+		value2: ProcessedDataItemTypes,
+	): boolean {
+		if (mode === 'latestIncrementalKey') {
+			const num1 = Number(value1);
+			const num2 = Number(value2);
+			if (!isNaN(num1) && !isNaN(num2)) {
+				return num1 > num2;
+			}
+			throw new ApplicationError(
```

Comment:
IMHO we should avoid using the `ApplicationError` directly but others might have other opinions in this regard. Maybe we could use a more specific error for this use case? E.g. `ProcessedDataError`

Reply:
I agree, I created an error class for these ones